library/core/src/main/java/com/google/android/exoplayer2/mediacodec/MediaCodecSelector.java

```diff
@@ -39,7 +39,7 @@
               MediaCodecUtil.getDecoderInfos(mimeType, requiresSecureDecoder);
           return decoderInfos.isEmpty()
               ? Collections.emptyList()
-              : Collections.singletonList(decoderInfos.get(0));
+              : decoderInfos;
```

Comment:
May be, assuming the first codec to the most compatible has to be checked with all attributes in MediaCodecSelector.  Let me check this again to make sure this is correct

Reply:
I could have used MediaCodecSelector.DEFAULT_WITH_FALLBACK. But I dont want to change DefaultRendererFactory. I will make the change